### PR TITLE
feature: qdrant filtering

### PR DESCRIPTION
**Pull request by densumesh**

### CODE REVIEW

File: server/src/data/schema.rs [hunk 1]
```diff
@@ -175,19 +175,6 @@ diesel::table! {
...
-        #[max_length = 256]
-        auth_name -> Nullable<Varchar>,
-        auth_srid -> Nullable<Int4>,
-        #[max_length = 2048]
-        srtext -> Nullable<Varchar>,
-        #[max_length = 2048]
-        proj4text -> Nullable<Varchar>,
-    }
-}
-
```
Comment (skeptrunedev) [inline server/src/data/schema.rs:190]: Idk why there is a diff here, but I assume MacOS?

File: server/src/lib.rs [hunk 2]
```diff
@@ -257,6 +260,51 @@ pub async fn main() -> std::io::Result<()> {
             log::info!("Failed to create collection: {:?}", err);
         });
 
+    let _ = qdrant_client
+        .create_field_index(
+            qdrant_collection.clone(),
+            "links",
```
Comment (skeptrunedev) [inline server/src/lib.rs:266]: Why `links` instead of `link`?

File: server/src/lib.rs [hunk 3]
```diff
@@ -257,6 +260,51 @@ pub async fn main() -> std::io::Result<()> {
...
+            None,
+        )
+        .await
+        .map_err(|err| {
+            log::info!("Failed to create index: {:?}", err);
+        });
+
+    let _ = qdrant_client
+        .create_field_index(
+            qdrant_collection.clone(),
```
Comment (skeptrunedev) [inline server/src/lib.rs:290]:
```suggestion
        });
    let _ = qdrant_client
        .create_field_index(
            qdrant_collection.clone(),
```

Comment (skeptrunedev) [reply server/src/lib.rs:290]: Nit

File: server/src/operators/card_operator.rs [hunk 4]
```diff
@@ -143,202 +143,129 @@ pub async fn search_card_query(
...
+    if !tag_set_inner.is_empty() {
+        filter
+            .must
+            .push(Condition::matches("tag_set", tag_set_inner));
     }
-    for link_url in link_inner.iter().skip(1) {
-        query = query.or_filter(card_metadata_columns::link.ilike(format!("%{}%", link_url)));
+
+    if !link_inner.is_empty() {
+        filter.must.push(Condition::matches("link", link_inner));
```
Comment (skeptrunedev) [inline server/src/operators/card_operator.rs:165]: Reasonably sure this means the index should be for `index` above

File: server/src/operators/card_operator.rs [hunk 5]
```diff
@@ -143,202 +143,129 @@ pub async fn search_card_query(
...
             }
         }
     }
 
     if let Some(quote_words) = parsed_query.quote_words {
         for word in quote_words.iter() {
-            query = query.filter(card_metadata_columns::content.ilike(format!("%{}%", word)));
+            filter
+                .must
+                .push(Condition::matches("card_html", word.clone() + " "));
```
Comment (skeptrunedev) [inline server/src/operators/card_operator.rs:252]:
```suggestion
                .push(Condition::matches("card_html", word.clone()));
```

I don't think you need the quote here. It's fine if it's a sub-word imo.

File: server/src/operators/qdrant_operator.rs [hunk 6]
```diff
@@ -11,19 +15,19 @@ pub async fn create_new_qdrant_point_query(
...
     author_id: Option<uuid::Uuid>,
 ) -> Result<(), actix_web::Error> {
     let qdrant = get_qdrant_connection()
         .await
         .map_err(|err| ServiceError::BadRequest(err.message.into()))?;
 
     let payload = match private {
-        true => {
-            json!({"private": true, "authors": vec![author_id.unwrap_or_default().to_string()]})
+        true => json!({"private": true, "authors": vec![author_id.unwrap_or_default().to_string()], "tag_set": card_metadata.tag_set.unwrap_or("".to_string()).split(',').collect_vec(), "link": card_metadata.link.unwrap_or("".to_string()).split(',').collect_vec(), "card_html": card_metadata.card_html.unwrap_or("".to_string()), "metadata": card_metadata.metadata.unwrap_or_default(), "timestamp": card_metadata.time_stamp.unwrap_or_default().timestamp()})
```
Comment (skeptrunedev) [inline server/src/operators/qdrant_operator.rs:26]: I don't think we need the `match private` anymore.

File: server/src/operators/card_operator.rs (hunk 4 above)
Comment (densumesh) [reply server/src/operators/card_operator.rs:165]: Wdym?

File: server/src/operators/card_operator.rs (hunk 5 above)
Comment (skeptrunedev) [reply server/src/operators/card_operator.rs:252]: >Yeah no there has to be a white space or else it doesn’t do a substring search

Per Matrix

File: server/src/operators/card_operator.rs (hunk 4 above)
Comment (skeptrunedev) [reply server/src/operators/card_operator.rs:165]: God dammit, I meant to write `link`

**Review by skeptrunedev — APPROVED** [review]

**Pull request merged**